### PR TITLE
Add /account/3pid/delete endpoint

### PR DESCRIPTION
**Pull request by dbkr**
Also fix a typo in a comment

### CODE REVIEW

File: synapse/handlers/auth.py [hunk 1]
```diff
@@ -617,6 +617,17 @@ def add_threepid(self, user_id, medium, address, validated_at):
             self.hs.get_clock().time_msec()
         )
 
+    @defer.inlineCallbacks
```
Comment (ara4n) [inline synapse/handlers/auth.py]: might be nice to fix the s/presenc e/presence/ typo on line 610

File: synapse/handlers/auth.py [hunk 2]
```diff
@@ -617,6 +617,17 @@ def add_threepid(self, user_id, medium, address, validated_at):
...
+    def delete_threepid(self, user_id, medium, address):
+        # 'Canonicalise' email addresses as per above
```
Comment (ara4n) [inline synapse/handlers/auth.py]: you're going to hate me for asking this, but: should we be canonicalising MSISDNs here? or are we assuming the client has already determined normal form?  (If so, a comment would be useful)

**Comment by ara4n** [comment]
lgtm other than 2 comments

File: synapse/handlers/auth.py (hunk 2 above)
Comment (dbkr) [inline synapse/handlers/auth.py]: In this case, the msisdn canonicalisation will have been done by the ID server (or the thing doing the verification). In fact, with us now having decided to have a separate thing to do the actual validation of 3pids, the email canonicalisation would go there too (it's a bit tricky for email given that squelching the localpart down to lowercase is not technically valid).

File: synapse/handlers/auth.py (hunk 1 above)
Comment (dbkr) [inline synapse/handlers/auth.py]: Fixed, although accidentally on develop.

**Pull request merged**